hub/core/chunk_engine/read.py

```diff
@@ -26,6 +32,7 @@ def sample_from_index_entry(
     return array_from_buffer(
```

Comment:
I know it is not part of this PR but `last_b_len = len(b)` can be moved out of for loop.
Doing this, will have no effect on code logic but should improve performance.

Reply:
b is being extended after each len() calculation